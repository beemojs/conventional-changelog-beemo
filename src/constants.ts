import { Group } from './types';

export const COMMIT_FORMAT_PREFIX =
	// eslint-disable-next-line unicorn/no-unsafe-regex
	/^(break|breaking|build|chore|ci|cd|deps|docs|feature|fix|internal|misc|new|patch|perf|release|revert|security|style|styles|test|tests|type|types|update)(?:\(([a-zA-Z0-9\-., ]+)\))?:/u;

export const GROUPS: Group[] = [
	{
		bump: 'major',
		emoji: '💥',
		label: 'Breaking',
		types: ['break', 'breaking'],
	},
	{
		bump: 'patch',
		emoji: '📦',
		label: 'Dependencies',
		types: ['deps'],
	},
	{
		emoji: '📘',
		label: 'Docs',
		types: ['docs'],
	},
	{
		bump: 'patch',
		emoji: '🐞',
		label: 'Fixes',
		types: ['fix', 'patch'],
	},
	{
		emoji: '🛠',
		label: 'Internals',
		types: ['ci', 'cd', 'build', 'test', 'tests', 'internal'],
	},
	{
		bump: 'patch',
		emoji: '📋',
		label: 'Misc',
		types: ['chore', 'misc'],
	},
	{
		bump: 'major',
		emoji: '🎉',
		label: 'Release',
		types: ['release'],
	},
	{
		bump: 'patch',
		emoji: '↩️',
		label: 'Reverts',
		types: ['revert'],
	},
	{
		bump: 'patch',
		emoji: '🔑',
		label: 'Security',
		types: ['security'],
	},
	{
		bump: 'patch',
		emoji: '🎨',
		label: 'Styles',
		types: ['style', 'styles'],
	},
	{
		bump: 'patch',
		emoji: '⚡️',
		label: 'Performance',
		types: ['perf'],
	},
	{
		bump: 'patch',
		emoji: '⚙️',
		label: 'Types',
		types: ['type', 'types'],
	},
	{
		bump: 'minor',
		emoji: '🚀',
		label: 'Updates',
		types: ['new', 'update', 'feature'],
	},
];
