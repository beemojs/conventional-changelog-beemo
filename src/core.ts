/**
 * @copyright   2019, Miles Johnson
 * @license     https://opensource.org/licenses/MIT
 */

export { checkCommitFormat } from './checkCommitFormat';
export { getTypeGroup } from './getTypeGroup';
export { parserOpts } from './parserOpts';
export { whatBump } from './whatBump';
export { writerOpts } from './writerOpts';
export * from './constants';
export * from './types';
