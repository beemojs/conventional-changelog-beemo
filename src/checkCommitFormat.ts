import { COMMIT_FORMAT_PREFIX } from './constants';
import type { CommitType } from './types';

export function checkCommitFormat(
	commit: string,
): { breaking: boolean; scope: string; type: CommitType } | null {
	const match = commit.match(COMMIT_FORMAT_PREFIX);

	if (!match) {
		return null;
	}

	const type = match[1] as CommitType;

	return {
		breaking: !!match[3] || type === 'breaking' || type === 'break',
		scope: match[2] || '',
		type,
	};
}
