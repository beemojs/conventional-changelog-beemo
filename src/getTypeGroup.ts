import { GROUPS } from './constants';
import type { CommitType, Group } from './types';

export function getTypeGroup(type: CommitType): Group {
	const group = GROUPS.find((g) => g.types.includes(type));

	if (!group) {
		throw new Error(`Cannot find group for type "${type}".`);
	}

	return group;
}
