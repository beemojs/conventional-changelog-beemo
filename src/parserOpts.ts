import { COMMIT_FORMAT_PREFIX } from './constants';
import { ParserOptions } from './types';

export const parserOpts: Partial<ParserOptions> = {
	headerCorrespondence: ['type', 'scope', 'message'],
	// Keep in sync with checkCommitFormat
	headerPattern: new RegExp(`^${COMMIT_FORMAT_PREFIX.source} (.*)$`, 'u'),
	mergeCorrespondence: ['pr', 'source'],
	mergePattern: /^Merged? pull request #(\d+) from (.*)/u,
	noteKeywords: 'Note',
	revertCorrespondence: [],
	revertPattern: /^Revert/u,
};
