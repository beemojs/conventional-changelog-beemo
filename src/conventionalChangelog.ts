import { parserOpts } from './parserOpts';
import { writerOpts } from './writerOpts';

export const conventionalChangelog = { parserOpts, writerOpts };
