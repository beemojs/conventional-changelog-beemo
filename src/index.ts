/**
 * @copyright   2019, Miles Johnson
 * @license     https://opensource.org/licenses/MIT
 */

import { parserOpts } from './parserOpts';
import { writerOpts } from './writerOpts';
import { whatBump } from './whatBump';

export * from './types';

export default function createPreset() {
	return {
		parser: parserOpts,
		writer: writerOpts,
		whatBump,
	};
}
