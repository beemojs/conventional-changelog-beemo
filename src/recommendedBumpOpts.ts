import getTypeGroup from './getTypeGroup';
import parserOpts from './parserOpts';
import { BumpOptions, Group, SemverLevel } from './types';

const options: BumpOptions = {
  parserOpts,

  whatBump(commits) {
    let level: SemverLevel = null;
    let breakings = 0;
    let features = 0;

    commits.forEach((commit) => {
      let group: Group;

      try {
        group = getTypeGroup(commit.type);
      } catch {
        return;
      }

      if (group.bump === 'major') {
        breakings += 1;
        level = 0;
      } else if (group.bump === 'minor') {
        features += 1;
        if (level === null || level === 2) {
          level = 1;
        }
      } else if (group.bump === 'patch') {
        if (level === null) {
          level = 2;
        }
      }
    });

    return {
      level,
      reason: `There are ${breakings} breaking changes and ${features} new features`,
    };
  },
};

export default options;
