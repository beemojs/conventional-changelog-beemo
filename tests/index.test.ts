/* eslint-disable jest/expect-expect, jest/no-test-callback */

import path from 'path';
import Stream from 'stream';
import conventionalChangelogCore from 'conventional-changelog-core';
import conventionalRecommendedBump from 'conventional-recommended-bump';
import shell from 'shelljs';
import { config } from '../src';

function gitDummyCommit(msg: string | string[], silent: boolean = true) {
  const args: string[] = ['--allow-empty', '--no-gpg-sign'];

  if (Array.isArray(msg)) {
    msg.forEach(m => {
      args.push(`-m"${m}"`);
    });
  } else {
    args.push(`-m"${msg}"`);
  }

  shell.exec(`git commit ${args.join(' ')}`, { silent });
}

function captureStreamOutput(stream: Stream.Readable, done: jest.DoneCallback) {
  let data = '';

  stream
    .on('error', (error: Error) => {
      done.fail(error);
    })
    .on('data', (chunk: string) => {
      data += String(chunk);
    })
    .on('end', () => {
      expect(data.trim()).toMatchSnapshot();
      done();
    });
}

describe('conventional-changelog-beemo', () => {
  const commonConfig = {
    config,
    pkg: {
      path: path.join(__dirname, 'package.json'),
    },
  };

  beforeEach(() => {
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    (shell.config as any).resetForTesting();
    shell.cd(__dirname);
    shell.mkdir('tmp');
    shell.cd('tmp');
    shell.mkdir('git-templates');
    shell.exec('git init --template=./git-templates');
  });

  afterEach(() => {
    shell.cd(__dirname);
    shell.rm('-rf', 'tmp');
  });

  it('supports all types at once', done => {
    gitDummyCommit(['release: New major!', 'Note: New build system.']);
    gitDummyCommit(['break: Forms have changed', 'Note: They are easier now!']);
    gitDummyCommit(['new: amazing new module', 'Not backward compatible.']);
    gitDummyCommit('fix: updated i18n');
    gitDummyCommit(['update(modal, button): added accessibility', 'closes #1, #2']);
    gitDummyCommit('feature(core): settings refactor');
    gitDummyCommit('Random commit with no type');
    gitDummyCommit('docs: added getting started');
    gitDummyCommit('style(button): polished rounded corners');
    gitDummyCommit(['security(auth): improved logic', 'fixes #3']);
    gitDummyCommit('Revert PR #1');
    gitDummyCommit('ci(travis): fixed yaml config');
    gitDummyCommit('build(deps): updated dev tools');
    gitDummyCommit('test: setup testing framework');
    gitDummyCommit('internal(ts): updated types');
    gitDummyCommit('deps(babel,jest): Bumped to latest');
    gitDummyCommit(['patch(router): Fix params']);
    gitDummyCommit('types: Removed any');

    captureStreamOutput(
      conventionalChangelogCore({
        ...commonConfig,
      }),
      done,
    );
  });

  it('works if there is no semver tag', done => {
    gitDummyCommit(['build: first build setup', 'Note: New build system.']);
    gitDummyCommit(['ci(travis): add TravisCI pipeline', 'Continuously integrated.']);
    gitDummyCommit(['new: amazing new module', 'Not backward compatible.']);
    gitDummyCommit(['fix(compile): avoid a bug', 'The Change is huge.']);
    gitDummyCommit(['update(ngOptions): make it faster', 'closes #1, #2']);
    gitDummyCommit('revert(ngOptions): bad commit');
    gitDummyCommit('fix(*): oops');

    captureStreamOutput(
      conventionalChangelogCore({
        ...commonConfig,
      }),
      done,
    );
  });

  it('works if there is a semver tag', done => {
    shell.exec('git tag v1.0.0');
    gitDummyCommit('update: some more features');

    captureStreamOutput(
      conventionalChangelogCore({
        ...commonConfig,
        outputUnreleased: true,
      }),
      done,
    );
  });

  it('works with unknown host', done => {
    gitDummyCommit('docs: add manual');

    captureStreamOutput(
      conventionalChangelogCore({
        ...commonConfig,
        pkg: {
          path: path.join(__dirname, 'package-unknown-repo.json'),
        },
      }),
      done,
    );
  });

  it('uses h1 for major versions', done => {
    gitDummyCommit('break: new shit');
    gitDummyCommit('release: new stuff');
    gitDummyCommit('fix: just a patch');

    captureStreamOutput(
      conventionalChangelogCore({
        ...commonConfig,
      }),
      done,
    );
  });

  it('uses h2 for minor versions', done => {
    gitDummyCommit('new: new shit');
    gitDummyCommit('update: new stuff');
    gitDummyCommit('feature(modal): better modals');
    gitDummyCommit('fix: just a patch');

    captureStreamOutput(
      conventionalChangelogCore({
        ...commonConfig,
      }),
      done,
    );
  });

  it('uses h3 for patch versions', done => {
    gitDummyCommit('docs: add a manual');
    gitDummyCommit('patch: just a patch');

    captureStreamOutput(
      conventionalChangelogCore({
        ...commonConfig,
      }),
      done,
    );
  });

  it('replaces #[0-9]+ with issue URL', done => {
    gitDummyCommit(['new(awesome): fix #88']);

    captureStreamOutput(
      conventionalChangelogCore({
        ...commonConfig,
      }),
      done,
    );
  });

  it('replaces @username with GitHub user URL', done => {
    gitDummyCommit(['feature(awesome): issue brought up by @bcoe! on Friday']);

    captureStreamOutput(
      conventionalChangelogCore({
        ...commonConfig,
      }),
      done,
    );
  });

  it('doesnt replace @username if wrapped in backticks', done => {
    gitDummyCommit(['deps: Updated \\`@types\\` packages.']);

    captureStreamOutput(
      conventionalChangelogCore({
        ...commonConfig,
      }),
      done,
    );
  });

  it('handles multiple notes', done => {
    gitDummyCommit(['release: Initial release', 'Note: Made a lot of changes']);
    gitDummyCommit(['fix(button): Made button changes', 'Note: Button is more buttony']);

    captureStreamOutput(
      conventionalChangelogCore({
        ...commonConfig,
      }),
      done,
    );
  });

  it('supports non public GitHub repository locations', done => {
    gitDummyCommit(['update(events): implementing #5 by @dlmr', ' closes #10']);
    gitDummyCommit('new: why this work?');

    captureStreamOutput(
      conventionalChangelogCore({
        ...commonConfig,
        pkg: {
          path: path.join(__dirname, 'package-custom-repo.json'),
        },
      }),
      done,
    );
  });

  it('only replaces with link to user if it is an username', done => {
    gitDummyCommit(['fix: use npm@5 (@username)']);
    gitDummyCommit([
      'build(deps): bump @dummy/package from 7.1.2 to 8.0.0',
      'break: The Change is huge.',
    ]);

    captureStreamOutput(
      conventionalChangelogCore({
        ...commonConfig,
      }),
      done,
    );
  });

  it('handles merge commits', done => {
    gitDummyCommit(['fix: use yarn']);
    gitDummyCommit('Merge pull request #29 from owner/repo');

    captureStreamOutput(
      conventionalChangelogCore({
        ...commonConfig,
      }),
      done,
    );
  });

  it('handles revert type', done => {
    gitDummyCommit('revert(foo): undo this');
    gitDummyCommit('Revert this is the PR title');

    captureStreamOutput(
      conventionalChangelogCore({
        ...commonConfig,
      }),
      done,
    );
  });

  describe('recommended bump', () => {
    ['break', 'breaking', 'release'].forEach(major => {
      it(`bumps major version for ${major}`, done => {
        gitDummyCommit(`${major}: new stuff`);
        gitDummyCommit(`${major}(todo): with scope`);

        conventionalRecommendedBump(
          {
            ...commonConfig,
          },
          (error: Error | null, result: object) => {
            expect(error).toBeNull();
            expect(result).toEqual({
              level: 0,
              reason: 'There are 2 breaking changes and 0 new features',
              releaseType: 'major',
            });
            done();
          },
        );
      });
    });

    ['new', 'update', 'feature'].forEach(minor => {
      it(`bumps minor version for ${minor}`, done => {
        gitDummyCommit(`${minor}: new stuff`);
        gitDummyCommit(`${minor}(todo): with scope`);

        conventionalRecommendedBump(
          {
            ...commonConfig,
          },
          (error: Error | null, result: object) => {
            expect(error).toBeNull();
            expect(result).toEqual({
              level: 1,
              reason: 'There are 0 breaking changes and 2 new features',
              releaseType: 'minor',
            });
            done();
          },
        );
      });
    });

    ['fix', 'deps', 'style', 'security', 'revert', 'misc', 'types'].forEach(patch => {
      it(`bumps patch version for ${patch}`, done => {
        gitDummyCommit(`${patch}: new stuff`);
        gitDummyCommit(`${patch}(todo): with scope`);

        conventionalRecommendedBump(
          {
            ...commonConfig,
            ignoreReverted: false,
          },
          (error: Error | null, result: object) => {
            expect(error).toBeNull();
            expect(result).toEqual({
              level: 2,
              reason: 'There are 0 breaking changes and 0 new features',
              releaseType: 'patch',
            });
            done();
          },
        );
      });
    });

    ['docs', 'ci', 'build', 'test', 'internal'].forEach(minor => {
      it(`doesnt bump version for ${minor}`, done => {
        gitDummyCommit(`${minor}: new stuff`);
        gitDummyCommit(`${minor}(todo): with scope`);

        conventionalRecommendedBump(
          {
            ...commonConfig,
          },
          (error: Error | null, result: object) => {
            expect(error).toBeNull();
            expect(result).toEqual({
              level: null,
              reason: 'There are 0 breaking changes and 0 new features',
            });
            done();
          },
        );
      });
    });

    it('does nothing when no type exist', done => {
      gitDummyCommit('new stuff');
      gitDummyCommit('commit without a type');

      conventionalRecommendedBump(
        {
          ...commonConfig,
        },
        (error: Error | null, result: object) => {
          expect(error).toBeNull();
          expect(result).toEqual({
            level: null,
            reason: 'There are 0 breaking changes and 0 new features',
          });
          done();
        },
      );
    });
  });
});
