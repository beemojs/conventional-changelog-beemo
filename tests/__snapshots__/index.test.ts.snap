// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`conventional-changelog-beemo > doesnt replace @username if wrapped in backticks 1`] = `
"## 1.0.0 (2024-05-08)

* deps: Updated \\ packages. ([48854c7](https://github.com/beemojs/conventional-changelog-beemo/commit/48854c7))"
`;

exports[`conventional-changelog-beemo > handles merge commits 1`] = `
"## 1.0.0 (2024-05-08)

* Merge pull request #29 from owner/repo ([5250ecb](https://github.com/beemojs/conventional-changelog-beemo/commit/5250ecb)), closes [#29](https://github.com/beemojs/conventional-changelog-beemo/issues/29)
* fix: use yarn ([f38a965](https://github.com/beemojs/conventional-changelog-beemo/commit/f38a965))"
`;

exports[`conventional-changelog-beemo > handles multiple notes 1`] = `
"## 1.0.0 (2024-05-08)

* fix(button): Made button changes ([1f3c84c](https://github.com/beemojs/conventional-changelog-beemo/commit/1f3c84c))
* release: Initial release ([bae1c06](https://github.com/beemojs/conventional-changelog-beemo/commit/bae1c06))"
`;

exports[`conventional-changelog-beemo > handles revert type 1`] = `
"## 1.0.0 (2024-05-08)

* Revert this is the PR title ([d6f662d](https://github.com/beemojs/conventional-changelog-beemo/commit/d6f662d))
* revert(foo): undo this ([884912a](https://github.com/beemojs/conventional-changelog-beemo/commit/884912a))"
`;

exports[`conventional-changelog-beemo > links commits/issues to deep repositories correctly 1`] = `
"## 1.0.0 (2024-05-08)

* update: supports sub-package links ([31fed38](https://github.com/beemojs/conventional-changelog-beemo/commit/31fed38)), closes [#10](https://github.com/beemojs/conventional-changelog-beemo/issues/10)"
`;

exports[`conventional-changelog-beemo > only replaces with link to user if it is an username 1`] = `
"## 1.0.0 (2024-05-08)

* build(deps): bump @dummy/package from 7.1.2 to 8.0.0 ([6c46583](https://github.com/beemojs/conventional-changelog-beemo/commit/6c46583))
* fix: use npm@5 (@username) ([608d4ac](https://github.com/beemojs/conventional-changelog-beemo/commit/608d4ac))"
`;

exports[`conventional-changelog-beemo > replaces #[0-9]+ with issue URL 1`] = `
"## 1.0.0 (2024-05-08)

* new(awesome): fix #88 ([029d4d2](https://github.com/beemojs/conventional-changelog-beemo/commit/029d4d2)), closes [#88](https://github.com/beemojs/conventional-changelog-beemo/issues/88)"
`;

exports[`conventional-changelog-beemo > replaces @username with GitHub user URL 1`] = `
"## 1.0.0 (2024-05-08)

* feature(awesome): issue brought up by @bcoe! on Friday ([b80bbae](https://github.com/beemojs/conventional-changelog-beemo/commit/b80bbae))"
`;

exports[`conventional-changelog-beemo > supports all types at once 1`] = `
"## 1.0.0 (2024-05-08)

* perf: Speeeeed ([578ef37](https://github.com/beemojs/conventional-changelog-beemo/commit/578ef37))
* types: Removed any ([98a5b32](https://github.com/beemojs/conventional-changelog-beemo/commit/98a5b32))
* patch(router): Fix params ([23d24a5](https://github.com/beemojs/conventional-changelog-beemo/commit/23d24a5))
* deps(babel,jest): Bumped to latest ([9c5a63a](https://github.com/beemojs/conventional-changelog-beemo/commit/9c5a63a))
* Random commit with no type ([8cfe89e](https://github.com/beemojs/conventional-changelog-beemo/commit/8cfe89e))
* Revert PR #1 ([bd556e5](https://github.com/beemojs/conventional-changelog-beemo/commit/bd556e5)), closes [#1](https://github.com/beemojs/conventional-changelog-beemo/issues/1)
* update(modal, button): added accessibility ([afff546](https://github.com/beemojs/conventional-changelog-beemo/commit/afff546)), closes [#1](https://github.com/beemojs/conventional-changelog-beemo/issues/1) [#2](https://github.com/beemojs/conventional-changelog-beemo/issues/2)
* internal(ts): updated types ([dcd1f00](https://github.com/beemojs/conventional-changelog-beemo/commit/dcd1f00))
* test: setup testing framework ([758f4fa](https://github.com/beemojs/conventional-changelog-beemo/commit/758f4fa))
* build(deps): updated dev tools ([3679834](https://github.com/beemojs/conventional-changelog-beemo/commit/3679834))
* ci(travis): fixed yaml config ([1efa186](https://github.com/beemojs/conventional-changelog-beemo/commit/1efa186))
* security(auth): improved logic ([1f1dd95](https://github.com/beemojs/conventional-changelog-beemo/commit/1f1dd95)), closes [#3](https://github.com/beemojs/conventional-changelog-beemo/issues/3)
* style(button): polished rounded corners ([97efdbc](https://github.com/beemojs/conventional-changelog-beemo/commit/97efdbc))
* docs: added getting started ([aaa4480](https://github.com/beemojs/conventional-changelog-beemo/commit/aaa4480))
* feature(core): settings refactor ([a7d9f6b](https://github.com/beemojs/conventional-changelog-beemo/commit/a7d9f6b))
* fix: updated i18n ([5cba2e8](https://github.com/beemojs/conventional-changelog-beemo/commit/5cba2e8))
* new: amazing new module ([6c1a592](https://github.com/beemojs/conventional-changelog-beemo/commit/6c1a592))
* break: Forms have changed ([d669e1b](https://github.com/beemojs/conventional-changelog-beemo/commit/d669e1b))
* release: New major! ([8c49b13](https://github.com/beemojs/conventional-changelog-beemo/commit/8c49b13))"
`;

exports[`conventional-changelog-beemo > supports non public GitHub repository locations 1`] = `
"## 3.0.0 (2024-05-08)

* new: why this work? ([6087d38](https://github.internal.example.com/conventional-changelog/internal/commit/6087d38))
* update(events): implementing #5 by @dlmr ([6c602a9](https://github.internal.example.com/conventional-changelog/internal/commit/6c602a9)), closes [#5](https://github.internal.example.com/conventional-changelog/internal/issues/5) [#10](https://github.internal.example.com/conventional-changelog/internal/issues/10)"
`;

exports[`conventional-changelog-beemo > uses h1 for major versions 1`] = `
"## 1.0.0 (2024-05-08)

* fix: just a patch ([cc6d48a](https://github.com/beemojs/conventional-changelog-beemo/commit/cc6d48a))
* release: new stuff ([9bcd1d8](https://github.com/beemojs/conventional-changelog-beemo/commit/9bcd1d8))
* break: new shit ([173722f](https://github.com/beemojs/conventional-changelog-beemo/commit/173722f))"
`;

exports[`conventional-changelog-beemo > uses h2 for minor versions 1`] = `
"## 1.0.0 (2024-05-08)

* fix: just a patch ([ef07da3](https://github.com/beemojs/conventional-changelog-beemo/commit/ef07da3))
* feature(modal): better modals ([d78c349](https://github.com/beemojs/conventional-changelog-beemo/commit/d78c349))
* update: new stuff ([9fc6fa9](https://github.com/beemojs/conventional-changelog-beemo/commit/9fc6fa9))
* new: new shit ([6cf43fa](https://github.com/beemojs/conventional-changelog-beemo/commit/6cf43fa))"
`;

exports[`conventional-changelog-beemo > uses h3 for patch versions 1`] = `
"## 1.0.0 (2024-05-08)

* patch: just a patch ([4abcd53](https://github.com/beemojs/conventional-changelog-beemo/commit/4abcd53))
* docs: add a manual ([134bad3](https://github.com/beemojs/conventional-changelog-beemo/commit/134bad3))"
`;

exports[`conventional-changelog-beemo > works if there is no semver tag 1`] = `
"## 1.0.0 (2024-05-08)

* perf: Speeeeed ([6da20f6](https://github.com/beemojs/conventional-changelog-beemo/commit/6da20f6))
* tests: Added before hooks ([227479a](https://github.com/beemojs/conventional-changelog-beemo/commit/227479a))
* type: Added unknown ([fdb2076](https://github.com/beemojs/conventional-changelog-beemo/commit/fdb2076))
* fix(*): oops ([878f533](https://github.com/beemojs/conventional-changelog-beemo/commit/878f533))
* fix(compile): avoid a bug ([6fb3737](https://github.com/beemojs/conventional-changelog-beemo/commit/6fb3737))
* revert(ngOptions): bad commit ([e077c22](https://github.com/beemojs/conventional-changelog-beemo/commit/e077c22))
* update(ngOptions): make it faster ([89bca64](https://github.com/beemojs/conventional-changelog-beemo/commit/89bca64)), closes [#1](https://github.com/beemojs/conventional-changelog-beemo/issues/1) [#2](https://github.com/beemojs/conventional-changelog-beemo/issues/2)
* new: amazing new module ([51a92ca](https://github.com/beemojs/conventional-changelog-beemo/commit/51a92ca))
* ci(travis): add TravisCI pipeline ([63f9639](https://github.com/beemojs/conventional-changelog-beemo/commit/63f9639))
* build: first build setup ([ce71419](https://github.com/beemojs/conventional-changelog-beemo/commit/ce71419))"
`;

exports[`conventional-changelog-beemo > works with unknown host 1`] = `
"## 1.0.0 (2024-05-08)

* docs: add manual ([2d98bf9](https://example.com/commits/2d98bf9))"
`;
